sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -428,16 +438,19 @@ def test_incr_mean_variance_axis_dim_mismatch(sparse_constructor):
             sp.random(5, 2, density=0.8, format="csr", random_state=0),
             sp.random(13, 2, density=0.8, format="csr", random_state=0),
         ),
-        (
-            sp.random(5, 2, density=0.8, format="csr", random_state=0),
-            sp.hstack(
-                [
-                    sp.csr_matrix(np.full((13, 1), fill_value=np.nan)),
-                    sp.random(13, 1, density=0.8, random_state=42),
-                ],
-                format="csr",
-            ),
-        ),
+        *[
+            (
+                sp.random(5, 2, density=0.8, format="csr", random_state=0),
+                sp.hstack(
+                    [
+                        csr_container(np.full((13, 1), fill_value=np.nan)),
+                        sp.random(13, 1, density=0.8, random_state=42),
+                    ],
+                    format="csr",
+                ),
+            )
+            for csr_container in CSR_CONTAINERS
```

Comment:
Due to the call to `sp.hstack`, we whill always return a sparse matrix.
The random in the previous tuple is also only testing sparse matrices.

One potential solution is to create sparse matrices, and parametrize with `CSR_CONTAINER` and intenally cast `X1` and `X2`:

```python
X1 = csr_container(X1)
X2 = csr_container(X2)
```